### PR TITLE
SNOW-368354 Support build and test for other architecture on linux

### PR DESCRIPTION
**Pull request by sfc-gh-cshi**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-368354. This fix is to allow build and test scripts pick right docker image and download required component(e.g. gosu) accordingly.

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.
    ci/build_docker.sh and ci/test_docker.sh could pick up right docker image with right gosu downloaded for both amd64 and arm64.

### CODE REVIEW

**Comment by sfc-gh-stakeda** [comment]
LGTM

**Review by sfc-gh-kwagner — APPROVED** [review]

File: ci/docker/connector_build/Dockerfile [hunk 1]
```diff
@@ -1,9 +1,11 @@
 # We use manylinux1 base image because pyarrow_manylinux2010 has a bug and wheel failed to be audited
-ARG BASE_IMAGE=quay.io/pypa/manylinux2010_x86_64
+ARG BASE_IMAGE
```
Comment (sfc-gh-mkeller) [inline ci/docker/connector_build/Dockerfile:2]: Please keep the default values
```suggestion
ARG BASE_IMAGE=quay.io/pypa/manylinux2010_x86_64
```

File: ci/docker/connector_build/Dockerfile [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
 FROM $BASE_IMAGE
 
 # This is to solve permission issue, read https://denibertovic.com/posts/handling-permissions-with-docker-volumes/
-RUN curl -o /usr/local/bin/gosu -SL "https://github.com/tianon/gosu/releases/download/1.11/gosu-amd64"
+ARG GOSU_URL
```
Comment (sfc-gh-mkeller) [inline ci/docker/connector_build/Dockerfile:6]: Default to old values
```suggestion
ARG GOSU_URL=https://github.com/tianon/gosu/releases/download/1.11/gosu-amd64
```

File: ci/docker/connector_test/Dockerfile [hunk 3]
```diff
@@ -3,13 +3,14 @@ ARG BASE_IMAGE
 FROM $BASE_IMAGE
 
 # This is to solve permission issue, read https://denibertovic.com/posts/handling-permissions-with-docker-volumes/
-RUN curl -o /usr/local/bin/gosu -SL "https://github.com/tianon/gosu/releases/download/1.11/gosu-amd64"
+ARG GOSU_URL
```
Comment (sfc-gh-mkeller) [inline ci/docker/connector_test/Dockerfile:6]: Keeping default values can make sure that we don't break downsstream project that might be using this Dockerfile
```suggestion
ARG GOSU_URL=https://github.com/tianon/gosu/releases/download/1.11/gosu-amd64
```

**Review by sfc-gh-mkeller — CHANGES_REQUESTED** [review]

**Review by sfc-gh-mkeller — APPROVED** [review]
Thanks @sfc-gh-cshi , looks good!

**Pull request merged**